### DISCUSSION TITLE
Plan for user access to recovery keys on main PDS, apps

### DISCUSSION THREAD

**Discussion by snarfed** [bluesky-social/atproto#1151] [Clarifications and "Why?"] [answered]
Hi all! This is low priority, not urgent, just filing since I'm curious and wanted somewhere to track. Do you know whether/how you plan to give current users access to their recovery keys? Bonus points if the answer doesn't include the terms "ASCII armor" or "openssl CLI" or "memorize this simple 26-word phrase." :trollface:

Related: #87, #474, #263 

Thanks in advance. Definitely low priority. Hope you all are keeping your heads afloat and staying sane. Velocity over the last month or so has been really impressive btw, everyone's having a ball playing with the toys you've given us and watching your progress!

**Comment by bnewbold** [comment, marked as answer]
This is pretty core to how `did:plc` works, but hasn't been exposed it in the client yet.

The short answer is that we will let folks create and configure their own recovery keys via their PDS, at account creation, or any other time.

The longer answer is that the UX for this will need care, as this is a security-sensitive subject. It is also entangled in how account migration will work: we'll want to transfer any PDS-controlled recovery keys (remove old PDS, add new PDS), but not touch any account-controlled recovery keys. Realistically, we probably also need to build a tool to help folks actually do the recovery process, either in the regular app or external software.
